src/patternfly/components/CustomSearchInput/custom-search-input.scss

```diff
@@ -0,0 +1,120 @@
+.pf-c-custom-search-input {
+  // Base border
+  --pf-c-custom-search-input__text--before--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-custom-search-input__text--before--BorderColor: var(--pf-global--BorderColor--300);
+
+  // Bottom border
+  --pf-c-custom-search-input__text--after--BorderBottomWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-custom-search-input__text--after--BorderBottomColor: var(--pf-global--BorderColor--200);
+  --pf-c-custom-search-input--hover__text--after--BorderBottomColor: var(--pf-global--primary-color--100);
+  --pf-c-custom-search-input__text--focus-within--after--BorderBottomWidth: var(--pf-global--BorderWidth--md);
+  --pf-c-custom-search-input__text--focus-within--after--BorderBottomColor: var(--pf-global--primary-color--100);
+
+  // Search input
+  --pf-c-custom-search-input__text-input--PaddingTop: var(--pf-global--spacer--form-element);
+  --pf-c-custom-search-input__text-input--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-custom-search-input__text-input--PaddingBottom: var(--pf-global--spacer--form-element);
+  --pf-c-custom-search-input__text-input--PaddingLeft: var(--pf-global--spacer--xl);
+  --pf-c-custom-search-input__text-input--MinWidth: 6ch;
+
+  // Icon
+  --pf-c-custom-search-input__icon--Left: var(--pf-global--spacer--sm);
+  --pf-c-custom-search-input__icon--Color: var(--pf-global--Color--200);
+  --pf-c-custom-search-input__text--hover__icon--Color: var(--pf-global--Color--100);
+  --pf-c-custom-search-input__text-input--focus--icon--Color: var(--pf-global--Color--100);
+  --pf-c-custom-search-input__icon--TranslateY: -50%;
+
+  // Utilities
+  --pf-c-custom-search-input__utilities--MarginRight: var(--pf-global--spacer--sm);
+  --pf-c-custom-search-input__utilities--child--MarginLeft: var(--pf-global--spacer--xs);
+  --pf-c-custom-search-input__utilities--c-button--PaddingRight: var(--pf-global--spacer--xs);
+  --pf-c-custom-search-input__utilities--c-button--PaddingLeft: var(--pf-global--spacer--xs);
+
+  position: relative;
+  display: flex;
+  padding: var(--pf-c-custom-search-input--PaddingTop) var(--pf-c-custom-search-input--PaddingRight) var(--pf-c-custom-search-input--PaddingBottom) var(--pf-c-custom-search-input--PaddingLeft);
+
+  &:hover {
+    --pf-c-custom-search-input__text--after--BorderBottomColor: var(--pf-c-custom-search-input--hover__text--after--BorderBottomColor);
+  }
+}
+
+.pf-c-custom-search-input__text {
+  flex: 1;
+
+  &::before,
+  &::after {
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    pointer-events: none;
+    content: "";
+  }
+
+  &::before {
+    border: var(--pf-c-custom-search-input__text--before--BorderWidth) solid var(--pf-c-custom-search-input__text--before--BorderColor);
```

Comment:
why is the border not set on `pf-c-custom-search-input` ?

Reply:
The base border (not the bottom that changes on hover/focus) could be, but I kept the borders on the same element because it seems cleaner to do that. I don't feel strongly about it, if you think moving the base border to the parent container is better.

Reply:
Ok, I guess my question would be why is it cleaner that way ? I wonder if we will run into any problems down the road with adding more elements or something, so to me having the border on the parent container seems the most full-proof/ consistent

Reply:
Yep, that's a good argument. An argument for it being cleaner is just to keep all of the border changes on the same element, so we aren't overriding borders on one element with borders on another element. But I'll change it, agree it gives us more versatility to add extra things on top of things if that base border is on the top level component element.